oak-core/src/main/java/org/apache/jackrabbit/oak/query/QueryImpl.java

```diff
@@ -1041,8 +1041,14 @@ private SelectorExecutionPlan getBestSelectorExecutionPlan(
             QueryIndex index = queryIndexes.get(i);
             double minCost = index.getMinimumCost();
             if (minCost > bestCost) {
-                // Stop looking if the minimum cost is higher than the current best cost
-                break;
+                if (minCost - bestIndex.getMinimumCost() < .00001) {
+                    // Continue with cost evaluation if minimum cost of both indexes are same i.e both indexes are on par.
```

Comment:
The calculation and the description don't match in my view. The "correct" way would be:

    Math.abs(minCost - bestIndex.getMinimumCost()) < .00001

I know that in reality, getMinimumCost is strictly ascending so it's only a theoretical problem... But I think it's easier to understand the code - well, for me :-)

Reply:
Considering an example, let's say we have 3 index providers A,B,C with min cost as 2.1, 2.2 and 2.3. 
Now this loop would still exit on 2.2 , whereas it could be possible for Index provider B to actually serve the query in a better way (actual cost to be low compared to index provider A). 

I am not sure if there's a way we can actually solve this (without actually comparing all the index providers - which would then lead to slower execution).

Reply:
I think the logic to skip is good in a sense that we are not spending too much time just to evaluate cost for all indexes. One probable way to make sure that we compare right indexes is to make mincost for providers configurable.

Reply:
+1